iXBRLViewerPlugin/viewer/src/js/viewer.js

```diff
@@ -375,59 +385,66 @@ export class Viewer {
 
-                // We may have already created an IXNode for this ID from a -sec-ix-hidden
-                // element 
-                let ixn = this._ixNodeMap[vuid];
-                if (!ixn) {
-                    ixn = new IXNode(vuid, nodes, docIndex);
-                    this._ixNodeMap[vuid] = ixn;
-                }
-                if (inHidden) {
-                    ixn.isHidden = true;
-                    nodes.addClass("ixbrl-element-hidden");
-                }
-                if (isContinuation) {
-                    nodes.addClass("ixbrl-continuation");
-                }
-                else {
+                    let ixn = this._getOrCreateIXNode(vuid, nodes, docIndex);
+                    if (inHidden) {
+                        ixn.isHidden = true;
+                        nodes.addClass("ixbrl-element-hidden");
+                    }
                     this._docOrderItemIndex.addItem(vuid, docIndex);
-                }
-                if (isNonFraction) {
-                    nodes.addClass("ixbrl-element-nonfraction");
-                    if (n.hasAttribute('scale')) {
-                        const scale = Number(n.getAttribute('scale'));
-                        // Set scale if the value is a valid number and is not a redundant 0/"ones" scale.
-                        if (!Number.isNaN(scale) && scale !== 0) {
-                            ixn.scale = scale;
+                    if (isNonFraction) {
+                        nodes.addClass("ixbrl-element-nonfraction");
+                        if (n.hasAttribute('scale')) {
+                            const scale = Number(n.getAttribute('scale'));
+                            // Set scale if the value is a valid number and is not a redundant 0/"ones" scale.
+                            if (!Number.isNaN(scale) && scale !== 0) {
+                                ixn.scale = scale;
+                            }
                         }
                     }
-                }
-                if (isNonNumeric) {
-                    nodes.addClass("ixbrl-element-nonnumeric");
-                    if (n.hasAttribute('escape') && n.getAttribute('escape').match(/^(true|1)$/)) {
-                        ixn.escaped = true;
+                    if (isNonNumeric) {
+                        nodes.addClass("ixbrl-element-nonnumeric");
+                        if (n.hasAttribute('escape') && n.getAttribute('escape').match(/^(true|1)$/)) {
+                            ixn.escaped = true;
+                        }
+                    }
+                    if (isFootnote) {
+                        nodes.addClass("ixbrl-element-footnote");
+                        ixn.footnote = true;
                     }
                 }
-                if (isFootnote) {
-                    nodes.addClass("ixbrl-element-footnote");
-                    ixn.footnote = true;
+            }
+            else if (isContinuation) {
```

Comment:
Out of curiosity, why split `isContinuation` to its own block?  Is there some functional change I'm not seeing, or is it just for clarity?  If the code is the same except for ID handling, perhaps a small helper function would be useful, and perhaps we could split `isFootnote` into its own block too.

Reply:
The functional change that precipitated this was switching from checking for absence of `target` attribute to presence of `id` attribute.  The condition at the start of the block was already a bit mad and not really what we wanted: if it's the right element type, but doesn't have an ID it shouldn't be considered by the subsequent `else`s, it should be ignored.  And then the content of the block was mostly conditional anyway, so it seemed clearer to split it.

I've done a bit of refactoring in c1fd3f7 to move common code into the existing helper functions.

Footnotes have a bit more in common with facts, because they get included in the document order index, and they have their own ID associated with them.